sdk-extensions/autoconfigure/src/testFullConfig/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdkTest.java

```diff
@@ -31,6 +35,23 @@ void setUp() {
     GlobalEventEmitterProvider.resetForTest();
   }
 
+  @SuppressWarnings("ResultOfMethodCallIgnored")
+  @Test
+  void build_addMetricReaderCustomizerPrometheus() {
+    AutoConfiguredOpenTelemetrySdkBuilder builder = AutoConfiguredOpenTelemetrySdk.builder();
+
+    builder.addPropertiesSupplier(() -> singletonMap("otel.metrics.exporter", "prometheus"));
+    builder.addMetricReaderCustomizer(
+        (reader, config) -> {
+          assertThat(reader).isInstanceOf(PrometheusHttpServer.class);
+          return PrometheusHttpServer.builder().setPort(1234).build();
+        });
+    builder.build();
```

Comment:
Can you wrap the resulting `OpenTelemetrySdk` in a try with resources block so we can close the prometheus server and free the port after completion?